components/Tag/Tag.jsx

```diff
@@ -0,0 +1,96 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+
+function getClass(isHover) {
+  return classNames({
+    inline: true,
+    relative: true,
+    'c-tag': true,
+    'is-hover': isHover,
+  });
+}
+
+function getButtonClass(isHover, isProcessing) {
+  return classNames({
+    'c-tag--button': true,
+    'btn-teal': isHover,
+    'is-hover': isHover,
+    'btn-gray': !isHover,
+    'is-processing': isProcessing,
+  });
+}
+
+function getLinkClass(isRemoveHover) {
+  return classNames({
+    'c-tag--remove': true,
+    'icon--center': true,
+    'icon-x-white': true,
+    'icon--xxsmall': true,
+    'btn-red': isRemoveHover,
+    'btn-teal': !isRemoveHover,
+  });
+}
+
+function truncate(str, maxLength) {
+  return str.length > maxLength ? str.slice(0, maxLength).concat('...') : str;
+}
+
+class Tag extends Component {
+
+  constructor() {
+    super();
+    this.state = {
+      isHover: false,
+      isRemoveHover: false,
+    };
+    this.setHover = this.setHover.bind(this);
+    this.setRemoveHover = this.setRemoveHover.bind(this);
+  }
+
+  setHover(value) {
+    return () => this.setState({ isHover: value });
+  }
+
+  setRemoveHover(value) {
+    return () => this.setState({ isRemoveHover: value });
```

Comment:
Do these `setState` calls need to be wrapped in a return function? Couldn't `setRemoveHover` just call `this.setState`?